app/src/test/java/com/github/se/polyfit/viewmodel/qrCode/BarCodeCodeViewModelTest.kt

```diff
@@ -48,7 +48,13 @@ class BarCodeCodeViewModelTest {
     for (i in 1..REQUIRED_SCAN_COUNT) {
       viewModel.addId("123456")
     }
-    assertEquals(listOf("123456"), viewModel.listId.value)
+
+    var i = 0
+    while (listOf("123456") != viewModel.listId.value) {
+      i++
+      sleep(100)
+      assert(i < 100)
```

Comment:
Maybe a not a good idea to have a sleep. I think that we lost points during the last milestones do to that.